src/Microsoft.OpenApi/Models/OpenApiComponents.cs

```diff
@@ -336,19 +354,14 @@ private void RenderComponents(IOpenApiWriter writer)
         {
             var loops = writer.GetSettings().LoopDetector.Loops;
             writer.WriteStartObject();
-            if (loops.TryGetValue(typeof(OpenApiSchema), out List<object> schemas))
+            if (loops.TryGetValue(typeof(JsonSchema), out List<object> schemas))
             {
 
-                writer.WriteOptionalMap(
-                   OpenApiConstants.Schemas,
-                   Schemas,
-                   static (w, key, component) => {
-                       component.SerializeAsV31WithoutReference(w);
-                   });
+                writer.WriteRaw(JsonSerializer.Serialize(schemas));
```

Comment:
If we make the previously suggested change, this change can be reverted. This won't work because it will be missing the "schemas" property.